File: app/src/main/java/com/myself/todo/Database/AlbumRepository.java
package com.myself.todo.Database;

import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;
import android.database.SQLException;
import android.database.sqlite.SQLiteDatabase;

import com.myself.todo.Beans.Album;

public class AlbumRepository {

    private static final String nome_banco = "agenda.db";
    private SQLiteDatabase banco;
    private DadosOpenHelper bancoListaOpenHelper;
    private String tblname = "Album";

    public AlbumRepository(Context context) {
        bancoListaOpenHelper = new DadosOpenHelper(context);
    }

    public void abrir() throws SQLException {
        banco = bancoListaOpenHelper.getWritableDatabase();
    }

    public void fecha() {
        if (banco != null) banco.close();
    }

    public void inserir(Album album, String usuario) {
        String status = "N";
        ContentValues novaFoto = new ContentValues();
        if (album.getFotouri().equals("") || album.getDescription().equals("") || usuario == null || usuario.equals("")) {
            System.out.println("usuario nao informado");

            return;
        }
        novaFoto.put("FOTO", album.getFotouri());
        novaFoto.put("DESCRICAO", album.getDescription());
        novaFoto.put("STATUS ", status);
        novaFoto.put("USUARIO", usuario);
        abrir();
        banco.insert(tblname, null, novaFoto);
        fecha();
    }

    public void apagar(long id, String usuario) {
        abrir();
        banco.delete(tblname, "_id = " + id + " and USUARIO = ?", new String[]{usuario});
        fecha();


    }

    public void apagarAll(String usuario) {
        abrir();
        banco.delete(tblname, "USUARIO = ?", new String[]{usuario});
        fecha();


    }

    public void UpdateUser(String usuarioantigo, String usuarionovo) {
        ContentValues produtoAlterado = new ContentValues();
        produtoAlterado.put("USUARIO", usuarionovo);
        abrir();
        banco.update(tblname, produtoAlterado, "USUARIO = ? ", new String[]{usuarioantigo});
        fecha();



    }


    public void favoritar(long id, String usuario) {
        String favorito= "F";
        ContentValues FotoAlterada = new ContentValues();
        FotoAlterada.put("status", favorito);
        abrir();
        banco.update(tblname, FotoAlterada, "_id = " + id + " and USUARIO = ?", new String[]{usuario});
        fecha();

    }

    public void concluir(long id){
        String favorito= "C";
        ContentValues produtoAlterado = new ContentValues();
        produtoAlterado.put("status",favorito);
        abrir();
        banco.update(tblname,produtoAlterado,"_id = "+ id,null);
        fecha();

    }


    public void unfavoritar(long id, String usuario) {
        String favorito= "N";
        ContentValues FotoAlterada = new ContentValues();
        FotoAlterada.put("status", favorito);
        abrir();
        banco.update(tblname, FotoAlterada, "_id = " + id + " and USUARIO = ?", new String[]{usuario});
        fecha();

    }


    public Cursor obterFotos(String usuario) {
        return banco.query(tblname, null, "USUARIO = ?", new String[]{usuario}, null, null, "STATUS");
    }

    public Cursor obterFotosRecentes(String usuario) {
        return banco.query(tblname, null, "USUARIO = ?", new String[]{usuario}, null, null, "HORA");
    }


    public Cursor obterFavoritos(String usuario) {
        return banco.query(tblname, null, "USUARIO = ? AND STATUS = 'F' ", new String[]{usuario}, null, null, "FOTO");
    }

    public Cursor obterUmEvento(long id){
        return banco.query(tblname,null,"_id = "+id,null,null,null,"ITEM");
    }

    public int contar(String usuario) {
        abrir();
        int count;
        Cursor mCount = banco.query(tblname, null, "USUARIO = ? ", new String[]{usuario}, null, null, null);
        mCount.moveToFirst();
        if (mCount.getCount() == 0) {
            count = 0;
            return count;
        } else {
            count = mCount.getInt(0);
            mCount.close();
            fecha();
            return count;
        }
    }

    public Album criafoto(Cursor resultado) {

        Album album = new Album();
        if(resultado.getCount() == 0){


            return null;
        }

        String uri = resultado.getString(resultado.getColumnIndexOrThrow("FOTO"));
        String descricao = resultado.getString(resultado.getColumnIndexOrThrow("DESCRICAO"));
        String dia = resultado.getString(resultado.getColumnIndexOrThrow("DIA"));
        String status = resultado.getString(resultado.getColumnIndexOrThrow("STATUS"));
        int id = resultado.getInt(resultado.getColumnIndexOrThrow("_id"));

        album.setId(String.valueOf(id));
        album.setDia(dia);
        album.setFotouri(uri);
        album.setDescription(descricao);
        album.setStatus(status);

        return album;



    }
}

File: app/src/main/java/com/myself/todo/Database/AlbumRepository.kt
package com.myself.todo.Database

import android.content.ContentValues
import android.content.Context
import android.database.Cursor
import android.database.SQLException
import android.database.sqlite.SQLiteDatabase
import com.myself.todo.Beans.Album
import org.junit.runner.RunWith

class AlbumRepository(context: Context?) {
    private var banco: SQLiteDatabase? = null
    private val bancoListaOpenHelper: DadosOpenHelper?
    private val tblname: String? = "Album"
    @Throws(SQLException::class)
    fun abrir() {
        banco = bancoListaOpenHelper.getWritableDatabase()
    }

    fun fecha() {
        if (banco != null) banco.close()
    }

    fun inserir(album: Album?, usuario: String?) {
        val status = "N"
        val novaFoto = ContentValues()
        if (album.getFotouri() == "" || album.getDescription() == "" || usuario == null || usuario == "") {
            println("usuario nao informado")
            return
        }
        novaFoto.put("FOTO", album.getFotouri())
        novaFoto.put("DESCRICAO", album.getDescription())
        novaFoto.put("STATUS ", status)
        novaFoto.put("USUARIO", usuario)
        abrir()
        banco.insert(tblname, null, novaFoto)
        fecha()
    }

    fun apagar(id: Long, usuario: String?) {
        abrir()
        banco.delete(tblname, "_id = $id and USUARIO = ?", arrayOf(usuario))
        fecha()
    }

    fun apagarAll(usuario: String?) {
        abrir()
        banco.delete(tblname, "USUARIO = ?", arrayOf(usuario))
        fecha()
    }

    fun UpdateUser(usuarioantigo: String?, usuarionovo: String?) {
        val produtoAlterado = ContentValues()
        produtoAlterado.put("USUARIO", usuarionovo)
        abrir()
        banco.update(tblname, produtoAlterado, "USUARIO = ? ", arrayOf(usuarioantigo))
        fecha()
    }

    fun favoritar(id: Long, usuario: String?) {
        val favorito = "F"
        val FotoAlterada = ContentValues()
        FotoAlterada.put("status", favorito)
        abrir()
        banco.update(tblname, FotoAlterada, "_id = $id and USUARIO = ?", arrayOf(usuario))
        fecha()
    }

    fun concluir(id: Long) {
        val favorito = "C"
        val produtoAlterado = ContentValues()
        produtoAlterado.put("status", favorito)
        abrir()
        banco.update(tblname, produtoAlterado, "_id = $id", null)
        fecha()
    }

    fun unfavoritar(id: Long, usuario: String?) {
        val favorito = "N"
        val FotoAlterada = ContentValues()
        FotoAlterada.put("status", favorito)
        abrir()
        banco.update(tblname, FotoAlterada, "_id = $id and USUARIO = ?", arrayOf(usuario))
        fecha()
    }

    fun obterFotos(usuario: String?): Cursor? {
        return banco.query(tblname, null, "USUARIO = ?", arrayOf(usuario), null, null, "STATUS")
    }

    fun obterFotosRecentes(usuario: String?): Cursor? {
        return banco.query(tblname, null, "USUARIO = ?", arrayOf(usuario), null, null, "HORA")
    }

    fun obterFavoritos(usuario: String?): Cursor? {
        return banco.query(tblname, null, "USUARIO = ? AND STATUS = 'F' ", arrayOf(usuario), null, null, "FOTO")
    }

    fun obterUmEvento(id: Long): Cursor? {
        return banco.query(tblname, null, "_id = $id", null, null, null, "ITEM")
    }

    fun contar(usuario: String?): Int {
        abrir()
        val count: Int
        val mCount = banco.query(tblname, null, "USUARIO = ? ", arrayOf(usuario), null, null, null)
        mCount.moveToFirst()
        return if (mCount.count == 0) {
            count = 0
            count
        } else {
            count = mCount.getInt(0)
            mCount.close()
            fecha()
            count
        }
    }

    fun criafoto(resultado: Cursor?): Album? {
        val album = Album()
        if (resultado.getCount() == 0) {
            return null
        }
        val uri = resultado.getString(resultado.getColumnIndexOrThrow("FOTO"))
        val descricao = resultado.getString(resultado.getColumnIndexOrThrow("DESCRICAO"))
        val dia = resultado.getString(resultado.getColumnIndexOrThrow("DIA"))
        val status = resultado.getString(resultado.getColumnIndexOrThrow("STATUS"))
        val id = resultado.getInt(resultado.getColumnIndexOrThrow("_id"))
        album.id = id.toString()
        album.dia = dia
        album.fotouri = uri
        album.description = descricao
        album.status = status
        return album
    }

    companion object {
        private val nome_banco: String? = "agenda.db"
    }

    init {
        bancoListaOpenHelper = DadosOpenHelper(context)
    }
}